File: codeforces/54.sum_of_round_numbers/sorn.cpp
#include <iostream>
#include <cstring>
#include <cmath>
#include <vector>
/* Author: José Rodolfo (jric2002) */
using namespace std;
int main() {
  unsigned short int t, n, d, o;
  signed short int k;
  string number, character;
  vector<unsigned short int> c;
  cin >> t;
  for (unsigned short int i = 0; i < t; i++) {
    cin >> n;
    number = to_string(n);
    k = (number.size() - 1);
    for (unsigned short int j = 0; j < number.size() && k >= 0; j++) {
      character = number.at(k);
      d = stoi(character);
      o = d * pow(10, j);
      if (o != 0) {
        c.push_back(o);
      }
      k--;
    }
    cout << c.size() << endl;
    for (vector<unsigned short int>::iterator it = c.begin(); it != c.end(); it++) {
      cout << *it << " ";
    }
    cout << endl;
    c.clear();
  }
  return 0;
}

File: codeforces/54.sum_of_round_numbers_[1352A]/sorn.py
# Author: José Rodolfo (jric2002)
t = int(input())
for i in range(0, t):
  n = str(input())
  c = []
  k = len(n) - 1
  j = 0
  while ((j < len(n)) and (k >= 0)):
    d = int(n[k])
    o = d * pow(10, j)
    if (o != 0):
      c.append(o)
    k -= 1
    j += 1
  print(len(c))
  print(" ".join(list(map(str, c))))